src/components/MenuItems.js

```diff
@@ -0,0 +1,108 @@
+import { useState, useEffect, useRef } from 'react';
+import Dropdown from './Dropdown';
+
+import { Link } from 'react-router-dom';
+
+const MenuItems = ({ items, depthLevel }) => {
+  const [dropdown, setDropdown] = useState(false);
+
+  let ref = useRef();
+
+  useEffect(() => {
+    const handler = (event) => {
+      if (
+        dropdown &&
+        ref.current &&
+        !ref.current.contains(event.target)
+      ) {
+        setDropdown(false);
+      }
+    };
+    document.addEventListener('mousedown', handler);
+    document.addEventListener('touchstart', handler);
+    return () => {
+      // Cleanup the event listener
+      document.removeEventListener('mousedown', handler);
+      document.removeEventListener('touchstart', handler);
+    };
+  }, [dropdown]);

```

Comment:
Why are you adding event listeners?
What happens if this useEffect was completely removed from the code?